Framework/Logger/include/Framework/Logger.h

```diff
@@ -17,17 +17,53 @@
 #include <fairlogger/Logger.h>
 #if __has_include(<fmt/format.h>)
 #include <fmt/format.h>
-#define LOGF(level, ...) LOG(level) << fmt::format(__VA_ARGS__)
+#include <fmt/printf.h>
+FMT_BEGIN_NAMESPACE
+template <typename S, typename Char = FMT_CHAR(S)>
+inline int vfprintf(fair::Logger& logger,
+                    const S& format,
+                    basic_format_args<typename basic_printf_context_t<
+                      internal::basic_buffer<Char>>::type>
+                      args)
+{
+  basic_memory_buffer<Char> buffer;
+  printf(buffer, to_string_view(format), args);
+  logger << buffer.data();
+  assert(*(buffer.data() + buffer.size()) == 0);
+  return static_cast<int>(buffer.size());
+}
+
+template <typename S, typename... Args>
+inline FMT_ENABLE_IF_T(internal::is_string<S>::value, int)
+  fprintf(fair::Logger& logger,
+          const S& format_str, const Args&... args)
+{
+  internal::check_format_string<Args...>(format_str);
+  typedef internal::basic_buffer<FMT_CHAR(S)> buffer;
+  typedef typename basic_printf_context_t<buffer>::type context;
+  format_arg_store<context, Args...> as{ args... };
+  return vfprintf(logger, to_string_view(format_str),
+                  basic_format_args<context>(as));
+}
+
+FMT_END_NAMESPACE
+
+#define LOGF(severity, ...)                                                                                                                                        \
+  for (bool fairLOggerunLikelyvariable = false; fair::Logger::Logging(fair::Severity::severity) && !fairLOggerunLikelyvariable; fairLOggerunLikelyvariable = true) \
```

Comment:
Looks good, I think here is a typo: fairLOgger... --> fairLogger.

Reply:
Yeah, but the typo is in FairLogger upstream, so I kept it just in case there is some logic behind... ;-)